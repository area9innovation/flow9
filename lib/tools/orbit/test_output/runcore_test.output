Flow compiler (3rd generation)

Processing 'orbit' on http server

Expected substring to get 3 args, but got 1 args.
WARNING: Core function substring is defined but call failed
Expected substring to get 3 args, but got 1 args.
Cannot call non-function value: substring
Sum: 12
Concatenated: Hello World
Array length: 1
ERROR: Expected int, not false
Comparison c (a > b): 0
ERROR: Expected int, not true
Comparison d (a < b): 0
ERROR: Expected int, not true
Comparison e (a == 5): 0
ERROR: Expected int, not false
Logical AND (c && d): 0
ERROR: Expected int, not true
Logical OR (c || d): 0
ERROR: Expected int, not true
Logical NOT (!c): 0
String length: 14
Substring: 7
Number to string: 42
String to number: 123
Nested calls: 50
Result: "Nested calls: 50"
