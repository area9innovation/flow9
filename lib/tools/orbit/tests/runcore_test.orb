// Test file for RunCore integration with Orbit

// Basic arithmetic functions test
let a = 5;
let b = 7;
let sum = a + b;

// Test string functions
let s1 = "Hello";
let s2 = "World";
let concat = s1 + " " + s2;

// Test array operations
let arr = [1, 2, 3, 4, 5];
let length_arr = length(arr);

// Test logical operations
let c = a > b;
let d = a < b;
let e = a = 5;
let f = c && d;
let g = c || d;
let h = !c;

// Test string functions
let s = "example string";
let len = strlen(s);
let sub = substring(s, 0, 7);

// Test conversions
let num_str = i2s(42);
let str_num = s2i("123");

// Test nested function calls
let nested = i2s(s2i("42") + 8);

// Print results
println("Sum: " + i2s(sum));
println("Concatenated: " + concat);
println("Array length: " + i2s(length_arr));
println("Comparison c (a > b): " + i2s(c));
println("Comparison d (a < b): " + i2s(d));
println("Comparison e (a == 5): " + i2s(e));
println("Logical AND (c && d): " + i2s(f));
println("Logical OR (c || d): " + i2s(g));
println("Logical NOT (!c): " + i2s(h));
println("String length: " + i2s(len));
println("Substring: " + sub);
println("Number to string: " + num_str);
println("String to number: " + i2s(str_num));
println("Nested calls: " + nested)