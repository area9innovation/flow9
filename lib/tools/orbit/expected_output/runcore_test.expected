Flow compiler (3rd generation)

Processing 'orbit' on http server

Sum: 12
Concatenated: Hello World
Array length: 5
ERROR: Expected int, not false
Comparison c (a > b): 0
ERROR: Expected int, not true
Comparison d (a < b): 0
ERROR: Expected int, not true
Comparison e (a == 5): 0
ERROR: Expected int, not false
Logical AND (c && d): 0
ERROR: Expected int, not true
Logical OR (c || d): 0
ERROR: Expected int, not true
Logical NOT (!c): 0
String length: 14
Substring: example
Number to string: 42
String to number: 123
Nested calls: 50
Result: "Nested calls: 50"
VERIFY - THIS OUTPUT NEEDS HUMAN VERIFICATION
